src/dune_engine/process.ml

```diff
@@ -480,6 +480,22 @@ let report_process_end stats common ~id =
   let event = Event.async (Int id) End common in
   Stats.emit stats event
 
+let report_process_event stats ~prog ~args
+    (resource_usage : Proc.resource_usage) =
+  let common =
+    let name = Filename.basename prog in
+    let ts = Timestamp.now () in
+    Event.common_fields ~cat:[ "process" ] ~name ~ts ()
+  in
+  let args =
+    [ ("process_args", `List (List.map args ~f:(fun arg -> `String arg))) ]
+  in
+  let dur =
```

Comment:
Conflating duration with CPU time is super-confusing. If we are able to record duration, we should do so, regardless of our ability to record the CPU time. Also, isn't stime a "system" CPU time, which is even more confusing because it's only a part of the total CPU time.

Reply:
Shouldn't we simply add optional fields for CPU time and keep recording duration as we did before? I would expect the code to be more uniform as well if we do that.

A separate reasonable improvement is to record duration more precisely (in the thread that calls wait), but that seems orthogonal.